volatility3/framework/plugins/windows/callbacks.py

```diff
@@ -155,6 +144,55 @@ def list_registry_callbacks(cls, context: interfaces.context.ContextInterface, l
             if callback.Function != 0:
                 yield "CmRegisterCallback", callback.Function, None
 
+    @classmethod
+    def _list_registry_callbacks_new(cls, context: interfaces.context.ContextInterface, layer_name: str, symbol_table: str,
+                                     callback_table_name: str) -> Iterable[Tuple[str, int, None]]:
+        """
+        Lists all registry callbacks via the CallbackListHead.
+        """
+
+        kvo = context.layers[layer_name].config['kernel_virtual_offset']
+        ntkrnlmp = context.module(symbol_table, layer_name = layer_name, offset = kvo)
+        full_type_name = callback_table_name + constants.BANG + "_CM_CALLBACK_ENTRY"
+
+        symbol_offset = ntkrnlmp.get_symbol("CallbackListHead").address
+        symbol_count_offset = ntkrnlmp.get_symbol("CmpCallBackCount").address
+
+        callback_count = ntkrnlmp.object(object_type = "unsigned int", offset = symbol_count_offset)
+
+        if callback_count == 0:
+            return
+
+        callback_list = ntkrnlmp.object(object_type = "_LIST_ENTRY", offset = symbol_offset)
+        for callback in callback_list.to_list(full_type_name, "Link"):
+            yield "CmRegisterCallbackEx", callback.Function, f"Alltitude: {callback.Alltitude.String}"
+
+    @classmethod
+    def list_registry_callbacks(cls, context: interfaces.context.ContextInterface, layer_name: str, symbol_table: str,
+                                callback_table_name: str) -> Iterable[Tuple[str, int, None]]:
+        """Lists all registry callbacks.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            layer_name: The name of the layer on which to operate
+            symbol_table: The name of the table containing the kernel symbols
+            callback_table_name: The nae of the table containing the callback symbols
+
+        Yields:
+            A name, location and optional detail string
+        """
+
+        kvo = context.layers[layer_name].config['kernel_virtual_offset']
+        ntkrnlmp = context.module(symbol_table, layer_name = layer_name, offset = kvo)
+
+        if ntkrnlmp.has_symbol("CmpCallBackVector") and ntkrnlmp.has_symbol("CmpCallBackCount"):
+            yield from cls._list_registry_callbacks_legacy(context, layer_name, symbol_table, callback_table_name)
+        elif ntkrnlmp.has_symbol("CallbackListHead") and ntkrnlmp.has_symbol("CmpCallBackCount"):
+            yield from cls._list_registry_callbacks_new(context, layer_name, symbol_table, callback_table_name)
+        else:
+            vollog.debug("Cannot find CmpCallBackVector or CmpCallBackCount or CallbackListHead")
```

Comment:
If we refine the conditional expression a little, wouldn't it be possible to inform the user more kindly about which Symbol does not exist? What do you think? 🙂

Reply:
I have added a debug log to print the state of those 3 symbols

Reply:
Thank you for considering my opinion, I hope it will help others.. 🙂